Create inheritance.md (#662)
Codecademy/docs: content/css/concepts/inheritance/inheritance.md

---
Title: 'Inheritance'
Description: 'The cascading in CSS. Styling that is applied to a chilled element when they are applied to its parent.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Inheritance'
  - 'Style'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
  - 'paths/full-stack-engineer-career-path'
---

CSS inheritance is the cascading in CSS. Inheritance is when styling that is applied to a chilled element when they are applied to its parent. Some properties are inherited but others are not.

## Inherided propities

<!-- prettier-ignore -->
| Property |
| --- |
| [`color`](https://www.codecademy.com/resources/docs/css/colors) |
| [`visibility`](https://www.codecademy.com/resources/docs/css/visibility) |
| All [Typograpgy](https://www.codecademy.com/resources/docs/css/typography) properties exept `text-decoration` and `text-overflow`. |
| [`cursor`](https://www.codecademy.com/resources/docs/css/cursor) |

## Non inherided propities

<!-- prettier-ignore -->
| Property |
| --- |

> | All [`background` properties](https://www.codecademy.com/resources/docs/css/background) |

Reviewer: ```suggestion
| All [background](https://www.codecademy.com/resources/docs/css/background) properties | Adds wallpaper visuals for HTML elements. | 
```
I would remove the backticks since this is linking to the CSS background concept and not the actual `background` property term entry. This suggestion also applies to some more rows down below.

> | [`border`](https://www.codecademy.com/resources/docs/css/borders) |

Reviewer: ```suggestion
| All [border](https://www.codecademy.com/resources/docs/css/borders) properties | Defines the border styles of an element. |
```

> | [`box-sizing`](https://www.codecademy.com/resources/docs/css/box-sizing) |
> | [`float`](https://www.codecademy.com/resources/docs/css/floats) |
> | All [`padding` properties](https://www.codecademy.com/resources/docs/css/padding) |
> | All [`grid` properties](https://www.codecademy.com/resources/docs/css/grids) |
> | All [`flex` properties](https://www.codecademy.com/resources/docs/css/flexbox) |
> | All [`overflow` properties](https://www.codecademy.com/resources/docs/css/overflow) |https://www.codecademy.com/resources/docs/css/background |
> | All [`margin` properties](https://www.codecademy.com/resources/docs/css/margins) |
> | All [`position` properties](https://www.codecademy.com/resources/docs/css/position) |
> | All [`sizing` properties](https://www.codecademy.com/resources/docs/css/sizing) |
> | All [`transition` properties](https://www.codecademy.com/resources/docs/css/transition) |
> | All [`transform` properties](https://www.codecademy.com/resources/docs/css/transform-functions/transform) |
> | [`text-decoration`](https://www.codecademy.com/resources/docs/css/typography/text-decoration) |
> | [`text-overflow`](https://www.codecademy.com/resources/docs/css/typography/text-overflow) |

Reviewer: ```suggestion
| [`box-sizing`](https://www.codecademy.com/resources/docs/css/box-sizing) | Includes `padding` and `border` in the calculation of an element's `height` and `width`. |
| All [Flexbox](https://www.codecademy.com/resources/docs/css/flexbox) properties | Arranges elements responsively and according to screen size. |
| [`float`](https://www.codecademy.com/resources/docs/css/floats) | Repositions an element to the left or right side of its containing element. |
| All [Grid](https://www.codecademy.com/resources/docs/css/grids) properties | Uses rows, columns, and gaps to organize content on a web page. |
| All [`margin` properties](https://www.codecademy.com/resources/docs/css/margins) | Sets the spacing beyond the border of an element. |
| All [overflow properties](https://www.codecademy.com/resources/docs/css/overflow) |https://www.codecademy.com/resources/docs/css/background | Manages content that spills  outside of its box. |
| All [padding](https://www.codecademy.com/resources/docs/css/padding) properties | Sets the spacing between the content and margin of an element. |
| All [position](https://www.codecademy.com/resources/docs/css/position) properties | Defines the placement of an element on a web page. |  
| All [sizing](https://www.codecademy.com/resources/docs/css/sizing) properties | Defines the `height` and `width` values of an element. |
| [`text-decoration`](https://www.codecademy.com/resources/docs/css/typography/text-decoration) | Styles the text of an element with line decorations. |
| [`text-overflow`](https://www.codecademy.com/resources/docs/css/typography/text-overflow) | Specifies how hidden content is signaled to the user. |
| [`transform`](https://www.codecademy.com/resources/docs/css/transform-functions/transform) |
| [`transform-origin`](https://www.codecademy.com/resources/docs/css/transform-functions/transform-origin) | Sets the position of a transformed element relative to its origin. |
| [`transform-style`](https://www.codecademy.com/resources/docs/css/transform-functions/transform-style) | Confirms whether an element's children are flattened or positioned in a 3D space. |
| All [transition](https://www.codecademy.com/resources/docs/css/transition) properties | Controls the animation speed and timing of property changes in an element. |
```

---
Title: 'Units'

> Description: 'The cascading in CSS. Styling that applies the the child element when the parent is the selector'

Reviewer 2: I think the phrase "when the parent is the selector" is making the sentence less clear. Maybe something more like: Styling that is applied to a chilled element when they are applied to its parent.

Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Inheritance'
  - 'Style'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
  - 'paths/full-stack-engineer-career-path'
---

## Inheritance

> CSS inheritance is the cascading in CSS. Inheritance is when styles are applied to children of a parent when the parent is the selector. Some properties are inherited but others are not.

Reviewer 2: See my prior comment.

Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Inheritance'
  - 'Style'
CatalogContent:
  - 'learn-css'
  - 'paths/front-end-engineer-career-path'
---

CSS inheritance describes how certain styles are initialized or computed depending on the CSS property and whether a value was set. Some properties are inherited with an initial, default value. Other properties are non-inherited and set to the computed value of its containing element by default.

## Inherited properties

Inherited properties apply to the element and its children.

| Property                                                                                    | Description                                                                                  |
| ------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------- |
| [`color`](https://www.codecademy.com/resources/docs/css/colors/color)                       | Sets the text color of an element.                                                           |

> | [`visibility`](https://www.codecademy.com/resources/docs/css/visibility)                    | Hides or reveals an element without affecting its placement within the page.                 |

Reviewer: Super nit-picky suggestion but could we make sure the rows in these tables are alphabetized by "Property", such as moving this row for `visibility` down below? (between `text-transform` and `word-spacing`)
